flink-kubernetes-operator-api/src/main/java/org/apache/flink/kubernetes/operator/api/status/CommonStatus.java

```diff
@@ -101,4 +108,67 @@ public ResourceLifecycleState getLifecycleState() {
      * loop immediately. For example autoscaler overrides have changed and we need to apply them.
      */
     @JsonIgnore @Internal private boolean immediateReconciliationNeeded = false;
+
+    public List<Condition> getConditions() {
+        switch (getLifecycleState()) {
+            case CREATED:
+                updateConditionIfNotExist(
+                        conditions,
+                        ConditionUtils.notReady(
+                                "The resource was created in Kubernetes but not yet handled by the operator"));
+                break;
+            case SUSPENDED:
+                updateConditionIfNotExist(
+                        conditions,
+                        ConditionUtils.notReady("The resource (job) has been suspended"));
+                break;
+            case UPGRADING:
+                updateConditionIfNotExist(
+                        conditions, ConditionUtils.notReady("The resource is being upgraded"));
+                break;
+            case DEPLOYED:
+                updateConditionIfNotExist(
+                        conditions,
+                        ConditionUtils.ready(
+                                "The resource is deployed, but it’s not yet considered to be stable and might be rolled back in the future"));
+                break;
+            case ROLLING_BACK:
+                updateConditionIfNotExist(
+                        conditions,
+                        ConditionUtils.notReady(
+                                "The resource is being rolled back to the last stable spec"));
+                break;
+            case ROLLED_BACK:
+                updateConditionIfNotExist(
+                        conditions,
+                        ConditionUtils.ready("The resource is deployed with the last stable spec"));
```

Comment:
This looks like a lot of duplicated code and copy-pasted strings from the `ResourceLifeCycleState` , I think we should add this conversion to the enum (or a utility class using the enum directly)